lib/Table/TableStyle.ts

```diff
@@ -471,19 +468,35 @@ export default class TableStyle {
     } else {
       // No column to color by, so use the same color for everything.
       let color: Color | undefined;
+
       const colorId = this.tableModel.uniqueId || this.tableModel.name;
-      if (colorTraits.nullColor) {
+
+      // If colorColumn is of type region - use regionColor
+      if (colorColumn?.type === TableColumnType.region && this.regionColor) {
+        color = this.regionColor;
+      } else if (colorTraits.nullColor) {
         color = Color.fromCssColorString(colorTraits.nullColor);
-      } else if (this.binColors.length > 0) {
-        color = this.binColors[0];
+      } else if (colorTraits.binColors && colorTraits.binColors.length > 0) {
+        color = Color.fromCssColorString(colorTraits.binColors[0]);
       } else if (colorId) {
         color = Color.fromCssColorString(getColorForId(colorId));
       }
 
```

Comment:
I changed this.binColors to `colorTraits.binColors` as I think we should only use binColors here if they are defined manually (not the default binColors).

This means that `Color.fromCssColorString(getColorForId(colorId))` is used more - which makes it easier to see different datasets